server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeVersionAllocationDecider.java

```diff
@@ -118,22 +118,22 @@ private static Decision isVersionCompatible(
         final RoutingNode target,
         final RoutingAllocation allocation
     ) {
-        if (target.node().getVersion().onOrAfter(recoverySource.version().toVersion())) {
+        if (target.node().getMaxIndexVersion().onOrAfter(recoverySource.version())) {
```

Comment:
I think this is unrelated to analysis changes?

Reply:
Oh yes, a stray unrelated change snuck in